lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -182,6 +182,30 @@ def get_cpu_facts(self, collected_facts=None):
             return cpu_facts
 
         cpu_facts['processor'] = []
+        cmd = self.module.get_bin_path('lscpu')
+        # cmd = cmd + " | egrep '^(Socket|Thread|Core|CPU|Model name)'"
+        rc, lscpu_out, err = self.module.run_command(cmd)
+        if lscpu_out:
+            for line in lscpu_out.splitlines():
+                data = line.split(' ')
+                if line.startswith('Socket(s)'):
+                    cpu_facts['processor_count'] = data[-1]
+                    continue
+                if line.startswith('Core(s) per socket'):
+                    cpu_facts['processor_cores'] = data[-1]
+                    continue
+                if line.startswith('Thread(s) per core'):
+                    cpu_facts['processor_threads_per_core'] = data[-1]
+                    continue
+                if line.startswith('Model name'):
+                    data = line.split(':')
+                    cpu_facts['processor'] = data[1].strip()
+                    continue
+            cpu_facts['processor_vcpus'] = int(cpu_facts['processor_count']) * int(cpu_facts['processor_cores']) * int(cpu_facts['processor_threads_per_core'])
+
+        if int(cpu_facts['processor_vcpus']) > 0:
```

Comment:
I'm not sure if we need this sanity check. If `lscpu` exists, we should trust it.

Reply:
removed this check